File: app/src/main/java/com/example/pearvideoclient/home/fragment/ContListAdapter.java
package com.example.pearvideoclient.home.fragment;

import androidx.annotation.Nullable;
import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.pearvideoclient.R;
import com.example.pearvideoclient.entity.RecommendBean;
import com.example.pearvideoclient.utils.GlideUtils;

import java.util.List;

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-22 16:44
 * @ClassName: ContListAdapter
 */
public class ContListAdapter extends BaseQuickAdapter<RecommendBean.DataListBean.ContListBeanX, BaseViewHolder> {
    public ContListAdapter(int layoutResId, @Nullable List<RecommendBean.DataListBean.ContListBeanX> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, RecommendBean.DataListBean.ContListBeanX item) {
        helper.setText(R.id.tv_video_name, item.getName())
                .setText(R.id.tv_video_author_duration, item.getUserInfo().getNickname() + " | " + item.getDuration());

        ImageView ivVideoImg = helper.getView(R.id.iv_video_img);
        GlideUtils.INSTANCE.loadWithPlaceHolder(item.getPic(), ivVideoImg, null, null);
        helper.addOnClickListener(R.id.rl_parent);
    }
}

File: app/src/main/java/com/example/pearvideoclient/home/fragment/ContListAdapter.kt
package com.example.pearvideoclient.home.fragment

import android.widget.ImageView

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.pearvideoclient.R
import com.example.pearvideoclient.entity.RecommendBean
import com.example.pearvideoclient.utils.GlideUtils

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-22 16:44
 * @ClassName: ContListAdapter
 */
class ContListAdapter(layoutResId: Int, data: List<RecommendBean.DataListBean.ContListBeanX>?) : BaseQuickAdapter<RecommendBean.DataListBean.ContListBeanX, BaseViewHolder>(layoutResId, data) {

    override fun convert(helper: BaseViewHolder, item: RecommendBean.DataListBean.ContListBeanX) {
        helper.setText(R.id.tv_video_name, item.name)
                .setText(R.id.tv_video_author_duration, item.userInfo.nickname + " | " + item.duration)

        val ivVideoImg = helper.getView<ImageView>(R.id.iv_video_img)
        GlideUtils.loadWithPlaceHolder(item.pic, ivVideoImg, null, null)
        helper.addOnClickListener(R.id.rl_parent)
    }
}
